src/java_bytecode/java_object_factory.cpp

```diff
@@ -582,15 +582,18 @@ exprt object_factory(
 
     exprt object=aux_symbol.symbol_expr();
 
-    gen_nondet_init(
-      object,
+    java_object_factoryt state(
       init_code,
+      !allow_null,
+      max_nondet_array_length,
       symbol_table,
-      loc,
+      message_handler);
+    state.gen_nondet_init_rec(
```

Comment:
No, I guess not. It was useful to be certain that the function was okay to remove. I'll add a commit to rename the remaining function back.